src/DBConnector/Connector.java

```diff
@@ -19,20 +20,46 @@
 
 public class Connector {
 
-   private static Statement myStmt;
+   //private static Statement myStmt;
+   private Connection myConn;
    private static String tableName = "UserAccount";
    private static final String dbName = "CodeU_2017DB";
    private static final String hostname = "ec2-176-34-225-252.eu-west-1.compute.amazonaws.com";
    private static final String DBusername = "group34";
    private static final String DBpassword = "codeu2017";
 
+   //the constructor can only create once per run.
+   private
+   PreparedStatement delete, getUsers, add, update, verify_selectPassword, selectUsername, dropAll, selectCount;
+
   public Connector() {
     try {
-      Connection myConn = DriverManager.getConnection(
+      myConn = DriverManager.getConnection(
           "jdbc:mysql://" + Connector.hostname + ":3306/" + Connector.dbName + "?useSSL=false",
           Connector.DBusername, Connector.DBpassword);
-      myStmt = myConn.createStatement();
-      
+
+      /*for deleteAccount()*/
+      delete = myConn.prepareStatement("DELETE  FROM "+tableName+" WHERE username = ?");
+
+      /*for updateAcount()*/
+      update = myConn.prepareStatement("UPDATE "+tableName+" SET password = ? WHERE username = ?");
+
+      /*for verifyAccount():*/
+      verify_selectPassword = myConn.prepareStatement("select password from "+tableName+" where username = ?");
+      //shared by both verify and add
+      selectUsername = myConn.prepareStatement("select username from "+tableName+" where username = ?");
+
+      /*for addUser():*/
+      add = myConn.prepareStatement("insert into "+tableName + "(username, password)" + "values(?,?)");
```

Comment:
This variable name is a bit misleading. Maybe `addStatement`? Just seeing `add` referenced later I found it confusing, especially since `add` is a common method for the `Collection` interface.